### PR TITLE
Upgrade to eslint v9 and move to neostandard

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:

This PR upgrades ESLint to version 9 and migrates our configuration to the new flat config system. Additionally, we've switched from eslint-config-standard to neostandard and introduced the lodash ESLint plugin for better code quality.

- Upgraded ESLint to version 9
- Migrated to the new `eslint.config.js` flat configuration format
- Replaced [eslint-config-standard](https://github.com/standard/eslint-config-standard) with [`neostandard ](https://github.com/neostandard/neostandard) for flat config compatibility
- Added [eslint-plugin-lodash](https://github.com/idok/eslint-plugin-lodash) for Lodash specific linting rules

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: frontend/eslint.config.cjs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        },
+        extensions: ['.js', '.vue'],
+      },
+    },
+  },
+  plugins: {
+    import: pluginImport,
+  },
+  rules: {
+    'import/order': ['error', {
```
Comment (petersutter) [inline frontend/eslint.config.cjs:41]:
```suggestion
  rules: {
    ...pluginImport.flatConfigs.recommended.rules,
    'import/order': ['error', {
```

**Review by petersutter — APPROVED** [review]
/lgtm

File: backend/eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      'no-debugger': 'error',
+    },
+  },
+  pluginSecurity.configs.recommended,
+  {
+    plugins: {
+      import: pluginImport,
+    },
+    rules: pluginImport.flatConfigs.recommended.rules,
+  },
```
Comment (petersutter) [inline backend/eslint.config.js:28]: just `pluginImport.flatConfigs.recommended`?

File: backend/eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const pluginLodash = require('eslint-plugin-lodash')
+const pluginImport = require('eslint-plugin-import')
+
+module.exports = [
+  ...neostandard({}),
+  {
+    rules: {
+      '@stylistic/comma-dangle': ['error', 'always-multiline'],
+      'no-console': 'error',
+      'no-debugger': 'error',
```
Comment (petersutter) [inline backend/eslint.config.js:19]: still need since we have maxWarnings=0?

File: backend/eslint.config.js (hunk 2 above)
Comment (holgerkoser) [reply backend/eslint.config.js:28]: The config sets `languageOptions.ecmaVersion` to `2018`. But we need `2022`

File: backend/eslint.config.js (hunk 3 above)
Comment (holgerkoser) [reply backend/eslint.config.js:19]: I have removed `no-debugger` since it is already part of the neostandard but we still need `no-console`

**Pull request merged**